includes/class-hours.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+namespace WPCW;
+
+if ( ! defined( 'ABSPATH' ) ) {
+
+	exit;
+
+}
+
+final class Hours extends Base_Widget {
+
+	/**
+	 * Widget constructor
+	 */
+	public function __construct() {
+
+		$widget_options = [
+			'classname'                   => 'wpcw-widgets wpcw-widget-hours',
+			'description'                 => __( 'Display your hours of operation.', 'contact-widgets' ),
+			'customize_selective_refresh' => true,
+		];
+
+		parent::__construct(
+			'wpcw_hours',
+			__( 'Hours of Operation', 'contact-widgets' ),
+			$widget_options
+		);
+
+	}
+
+	/**
+	 * Widget form fields
+	 *
+	 * @param array $instance The widget options
+	 *
+	 * @return string|void
+	 */
+	public function form( $instance ) {
+
+		parent::form( $instance );
+
+		$fields = $this->get_fields( $instance );
+
+		echo '<div class="wpcw-widget wpcw-widget-hours">';
+
+		echo '<div class="title">';
+
+		// Title field
+		$this->render_form_input( array_shift( $fields ) );
+
+		echo '</div>';
+
+		echo '<div class="form">';
+
+		foreach ( $fields as $key => $field ) {
+
+			if ( 'days' === $key ) {
+
+				parent::print_label( $field );
+
+				foreach ( $field['days'] as $day => $hours ) {
+
+					$this->render_day_input( $fields['days'], $day, $hours );
+
+				}
+
+				continue;
+
+			}
+
+			$method = $field['form_callback'];
+
+			if ( is_callable( [ $this, $method ] ) ) {
+
+				$this->$method( $field );
+
+			}
+
+		}
+
+		// Workaround customizer refresh @props @westonruter
+		echo '<input class="customizer_update" type="hidden" value="">';
+
+		echo '</div>'; // End form
+
+		echo '</div>'; // End wpcw-widget-hours
+
+	}
+
+	/**
+	 * Front-end display
+	 *
+	 * @param array $args
+	 * @param array $instance
+	 */
+	public function widget( $args, $instance ) {
+
+		$fields = $this->get_fields( $instance );
+
+		if ( $this->is_widget_empty( $fields ) ) {
+
+			return;
+
+		}
+
+		$this->before_widget( $args, $fields );
+
+		foreach ( $fields as $field ) {
+
+			if ( isset( $field['days'] ) ) {
+
+				foreach ( $field['days'] as $day_of_week => $store_hours ) {
+
+					$hours = $store_hours['not_open'] ? __( 'Closed', 'contact-widgets' ) : $store_hours['open'] . apply_filters( 'wpcw_hours_seperator', ' - ' ) . $store_hours['closed'];
+					$class = $store_hours['not_open'] ? 'closed' : 'open';
+
+					if ( $store_hours['custom_text_checkbox'] ) {
```

Comment:
It's worth noting here that when a user has a 'custom text' specified the micro formats are not added to the hours of operation item.